src/components/splitButton/splitButton.ts

```diff
@@ -0,0 +1,409 @@
+export class SplitButton extends LitElement {
+  static override styles = SplitButtonScss;
+
+  /** @ignore */
+  static override shadowRootOptions = {
+    ...LitElement.shadowRootOptions,
+    delegatesFocus: true,
+  };
+
+  /** ARIA label for the buttons for accessibility. */
+  @property({ type: String })
+  description = '';
+
+  /** Split button name. */
+  @property({ type: String })
+  name = '';
+
+  /** Specifies the visual appearance/kind of the split button. */
+  @property({ type: String })
+  kind: SPLIT_BTN_KINDS = SPLIT_BTN_KINDS.PRIMARY_APP;
+
+  /** Specifies the size of the split button. */
+  @property({ type: String })
+  size: SPLIT_BTN_SIZES = SPLIT_BTN_SIZES.MEDIUM;
+
+  /** Specifies the position of the icon relative to any split button text. Default `'left'`. This is optional and work with icon slot.*/
+  @property({ type: String })
+  iconPosition: SPLIIT_BTN_ICON_POSITION = SPLIIT_BTN_ICON_POSITION.LEFT;
+
+  /** Split button text (required) */
+  @property({ type: String })
+  label = '';
+
+  /** Determines if the split button is disabled. */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  /** Determines if the split button indicates a destructive action. */
+  @property({ type: Boolean, reflect: true })
+  destructive = false;
+
+  /** Menu CSS min-width value. */
+  @property({ type: String })
+  menuMinWidth = 'initial';
+
+  /** Listbox/menu open state. */
+  @property({ type: Boolean })
+  open = false;
+
+  /**
+   * Queries any slotted options.
+   * @ignore
+   */
+  @queryAssignedElements({ selector: 'kd-splitbutton-option' })
+  options!: Array<any>;
+
+  /**
+   * Open menu upwards.
+   * @ignore
+   */
+  @state()
+  _openUpwards = false;
+
+  /**
+   * Queries the .select DOM element.
+   * @ignore
+   */
+  @query('.select')
+  buttonEl!: HTMLElement;
+
+  /**
+   * Assistive text for screen readers.
+   * @ignore
+   */
+  @state()
+  assistiveText = 'Split button menu options.';
+
+  override render() {
+    const typeClassMap = {
+      [SPLIT_BTN_KINDS.PRIMARY_APP]: 'primary-app',
+      [SPLIT_BTN_KINDS.PRIMARY_WEB]: 'primary-web',
+      [SPLIT_BTN_KINDS.SECONDARY]: 'secondary',
+    };
+
+    const baseTypeClass = typeClassMap[this.kind];
+    const destructModifier = this.destructive ? '-destructive' : '';
+
+    const classes = {
+      button: true,
+      [`kd-split-btn--${baseTypeClass}${destructModifier}`]: true,
+      [`kd-split-btn--${baseTypeClass}`]: !this.destructive,
+      'kd-split-btn--large': this.size === SPLIT_BTN_SIZES.LARGE,
+      'kd-split-btn--small': this.size === SPLIT_BTN_SIZES.SMALL,
+      'kd-split-btn--medium': this.size === SPLIT_BTN_SIZES.MEDIUM,
+      [`kd-split-btn--icon-${this.iconPosition}`]: !!this.iconPosition,
+    };
+
+    return html`
+      <div
+        class="split-btn-wrapper ${this.open ? 'open' : ''}"
+        id="label-${this.name}"
+      >
+        <!-- label button UI -->
+        <button
```

Comment:
This button has no click handler now.

Reply:
yes no needed. As it only represent label

Reply:
That doesn't make sense. The visible action button must still be clickable.

Reply:
yeah but what is the action at @click ? e.preventDefault(); ?

Reply:
Done. emits origin event on this click